packages/InteractiveProfilingTool-Core.package/IPTStats.class/instance/formatGarbageCollectorStats.st

```diff
@@ -0,0 +1,17 @@
+formatting
+formatGarbageCollectorStats
+
+	| stats lines full incremental |
+	
+	stats := self root getGarbageCollectorStats.
+	
+	full := stats at: #full.
+	incremental := stats at: #incremental.
+	
+	lines := {
+		('Full GCs: {1} total, {2}ms ({3}% uptime)' format: full values).
+		('Incremental GCs: {1} total, {2}ms ({3}% uptime)' format: incremental values).
+		('Tenures: {1}' format: {stats at: #tenures}).
```

Comment:
Gibt es einen Grund, wieso in Line 14 und 15 die Collection direkt angegeben wird, während ihr oben für `full` und `incremental` Variablen gebaut habt? Wenn das geht, fände ich eine einheitlche Lösung schöner, ich habe aber keine Vorlieben welche.

edit: Wobei ich gerade sehe, dass zum Beispiel `formatMemoryStats` auch die Variante ohne Variable benutzt also vielleicht eher die.

Reply:
Das liegt daran, dass `stats at: #full`/`stats at: #incremental` Dctionarys sind, während `stats at: #tenures`/`stats at: #rootOverflows` Integers sind. Letzteres gilt auch für die in `formatMemoryStats`verwendeten Werte.

Reply:
https://github.com/hpi-swa-teaching/InteractiveProfilingTool/blob/41-show-stats/packages/InteractiveProfilingTool-Core.package/IPTMessageTallyWrapper.class/instance/getMemoryStats.st#L12

https://github.com/hpi-swa-teaching/InteractiveProfilingTool/blob/41-show-stats/packages/InteractiveProfilingTool-Core.package/IPTMessageTallyWrapper.class/instance/getGarbageCollectorStats.st#L21